File: google/codejam20/qual/ppr.py
t = int(input())
for tt in range(1,t+1):
    n = int(input())
    j = [True for _ in range(24*60+1)]
    c = [True for _ in range(24*60+1)]
    res = ""
    for _ in range(n):
        start,end = [int(k) for k in input().split(' ')]
        if res=="IMPOSSIBLE":
            continue
        jcan = True
        ccan = True
        for i in range(start,end):
            jcan = j[i] and jcan
            ccan = c[i] and ccan
            if not jcan and not ccan:
                break
        if not jcan and not ccan:
            res="IMPOSSIBLE"
            continue
        if jcan:
            res += 'J'
            for i in range(start,end):
                j[i] = False
        else:
            res += 'C'
            for i in range(start,end):
                c[i] = False
    print("Case #{}: {}".format(tt,res))

File: google/codejam20/qual/ppr.cpp
#include <bits/stdc++.h>
#define INF 2e18
#define MOD 10000007

using namespace std;

typedef unsigned long long ull;
typedef long long ll;
typedef pair<int,int> ii;

int t, s, e, n;

int main(){
    /* ios_base::sync_with_stdio(false); */
    cin.tie(NULL);

    cin>>t;
    for(int tt=0;tt<t;tt++) {
        cin>>n;
        bool j[60*24+5];
        bool c[60*24+5];
        string res = "";
        for(int i=0;i<60*24+5;i++) {
            j[i]=true;
            c[i]=true;
        }
        for(int nn=0;nn<n;nn++) {
            cin>>s>>e;
            if(res=="IMPOSSIBLE") {
                continue;
            }
            bool jcan=true, ccan=true;
            for(int i=s; i<e;i++) {
                jcan &= j[i];
                ccan &= c[i];
            }

            if(!(jcan || ccan)) {
                res="IMPOSSIBLE";
            } else if (jcan) {
                for(int i=s;i<e;i++) {
                    j[i] = false;
                }
                res.append("J");
            } else {
                for(int i=s;i<e;i++) {
                    c[i] = false;
                }
                res.append("C");
            }
        }
        printf("Case #%d: ", tt+1);
        cout<<res<<'\n';
    }


    return 0;
}
